src/shadowbox/server/server_access_key.ts

```diff
@@ -187,15 +182,14 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
   }
 
   async setAccessKeyQuota(id: AccessKeyId, quota: AccessKeyQuota): Promise<boolean> {
-    if (!quota || !quota.quota || !quota.window || quota.quota.bytes < 0 ||
-        quota.window.hours < 0) {
+    if (!quota || !quota.data || !quota.window || quota.data.bytes < 0 || quota.window.hours < 0) {
       return false;
     }
     const accessKey = this.getAccessKey(id);
     if (!accessKey) {
       return false;
     }
-    accessKey.quota = quota;
+    accessKey.quotaUsage = {quota, usage: {bytes: 0}};
     try {
       this.saveAccessKeys();
       const quotaStautsChanged = await this.updateAccessKeyQuotaStatus(accessKey);
```

Comment:
This is actually about the catch but github won't let me comment on that line.

Do we want to add some debug logging of the error or some similar thing?  It could be frustrating if an admin can't reset quotas and there's no log of the error.

Reply:
I think this is part of a larger issue that is the lack of typed errors in this class. Instead of adding logging, which wouldn't be propagated to the API call anyway, I propose we implement typed errors so that the caller can log/communicate the error accordingly.

Reply:
SGTM. can we add a quick todo?

Reply:
discussed offline.  all good.